### PR TITLE
Pubdev 4438 one off error

### PR DESCRIPTION
**Pull request by wendycwong**
Two things:

1. Fixed index problem in method index of AstNumList.java.  Given a list of [0,1,2,3, ...], the returned index for 0 and 1 are both 0 and the return index of 2 or more is one less that it should be.  Easy fix: changed 
                     if (bIdx >= 0) return water.util.ArrayUtils.sum(_cnts, 0, bIdx - 1);
TO
                     if (bIdx >= 0) return water.util.ArrayUtils.sum(_cnts, 0, bIdx );

2. Add Runit test to make sure the assignment error is gone. 

-------------------------
The reason that you only see this problem if you have more than 1000 elements is because for data with less than 1001 elements, different function calls are made!

### CODE REVIEW

File: h2o-core/src/main/java/water/rapids/ast/params/AstNumList.java [hunk 1]
```diff
@@ -241,7 +241,8 @@ public boolean has(long v) {
    */
   public long index(long v) {
     int bIdx = findBase(v);
-    if (bIdx >= 0) return water.util.ArrayUtils.sum(_cnts, 0, bIdx - 1);
+//    if (bIdx >= 0) return water.util.ArrayUtils.sum(_cnts, 0, bIdx - 1);
```
Comment (michalkurka) [inline h2o-core/src/main/java/water/rapids/ast/params/AstNumList.java]: please remove the commented out line

Add a junit test for this method (to test the edge cases)
